### ISSUE TITLE
Create Shard per Unique Partitioning Key

### ISSUE THREAD

**Issue by etiennedi** [weaviate/weaviate#2946]
TODO: Add more content

tl;dr
* When in MMT mode, there are no more virtual shards, only physical shards
* In MMT mode, the user must define a partitioning key
* Any time Weaviate sees a previously unseen value for the partitioning key, a new shard is created
* Any operation that requires access to a specific object must provide the partitioning key
* The goal is an MVP style implementation, we can scope out anything that makes this harder. For example, it's fine if not all API endpoints are supported or cross-refs aren't supported in the first iteration. The follow up issue to tackle anything we scoped out here is #2950
